src/sources/kubernetes_logs/mod.rs

```diff
@@ -445,10 +454,13 @@ impl Source {
 
         let mut stream = partial_events_merger.transform(Box::pin(events));
         let event_processing_loop = out.send_event_stream(&mut stream);
-
-        let mut lifecycle = Lifecycle::new();
+        let mut lc: Lifecycle;
+        match ordered {
+            true => lc = Lifecycle::new(lifecycle::LifecycleOrder::Ordered),
+            false => lc = Lifecycle::new(lifecycle::LifecycleOrder::Unordered),
```

Comment:
If you move the `ordered` parameter to `Lifecycle::new`, I think you can hide the `LifecycleOrder` enum entirely in that module and not have it exposed here. I would also recommend against shortening the name.

Reply:
@bruceg now into Lifecycle::new just passes an 'ordered' as arguments